### ISSUE TITLE
Escape dhall 'Text' variables with boolean values in YAML

### ISSUE THREAD

**Issue by chshersh** [dhall-lang/dhall-haskell#1785]
Hi, thanks for working on Dhall and all the awesome ecosystem!

We're using Dhall in Relude to generate thousands of lines of custom HLint suggestions from a much smaller Dhall config. However, I've started to have a problem when using `dhall-json-1.6.4` (the latest version). We reexport Haskell function with the name `on`, which, IIUC, parses as the boolean value `true` in YAML. As a consequence, this results in the following HLint error when I'm trying to use the generated config:

```
hlint: user error (Failed to read YAML configuration file /home/shersh/haskell/kowainik/relude/.hlint.yaml
  Aeson exception:
Error in $: Error when decoding YAML, Expected a String
Along path: root 479 warn rhs
When at: Bool
warn:
  lhs: Data.Function.on
  note: '''on'' is already exported from Relude'
  name: Use 'on' from Relude
  rhs: true
)
```

This started to happen after an attempt to use the latest version of `dhall-json`. Here is the diff for the relevant YAML piece:

```diff
 - warn:
     lhs: Data.Function.on
-    note: '''on'' is already exported from Relude'
-    name: Use 'on' from Relude
-    rhs: 'on'
+    name: "Use 'on' from Relude"
+    note: "'on' is already exported from Relude"
+    rhs: on
```

When I tried to replace the string `"on"` in my Dhall config with `"\"on\""`, I got the same string in the HLint config, which is also incorrect, so I don't see a way to work around this issue at the moment except removing this particular warning from `relude` (which is undesirable, as you can understand).

**Comment by sjakobi** [comment]
`aeson-yaml`, which `dhall-to-yaml` relies on, gives its special "boolean string" treatment only to the strings `"true"` and `"false"`. [`on`, `off`, `yes`, etc. aren't valid YAML booleans in YAML 1.2](https://stackoverflow.com/questions/42283732/are-on-and-off-supposed-to-be-interpreted-as-true-or-false-in-yaml-1-2) – they are simple strings.

So I think ideally `hlint` would update its YAML parser to conform with YAML 1.2.

**Comment by Gabriel439** [comment]
@sjakobi: It probably can't hurt to quote those keys anyway, just for backwards compatibility with YAML 1.1

**Comment by sjakobi** [comment]
Yeah, I agree. That might also help with compatibility with other consumers of  `dhall-to-yaml[-ng]` output.

**Comment by german1608** [comment]
I'm working on a fix for this

**Comment by chshersh** [comment]
I've installed the latest version of `dhall-to-yaml` (1.7.5) and I still see that the string `on` is not quoted. Maybe the regression was introduced again?

**Comment by sjakobi** [comment]
Looking at the PR that closed this issue (#1788) it appears that we didn't actually change the behaviour of `dhall-to-yaml` because the `aeson-yaml` maintainer saw more demand for compatibility with YAML 1.2 and prettier output.

We did update `dhall-to-yaml-ng` though:

```
$ echo '{ on = "on" }' | dhall-to-yaml-ng
'on': 'on'
```

So I'd suggest that you switch to using `dhall-to-yaml-ng` (from the `dhall-yaml` package) instead.
